pygeos/test/test_geometry.py

```diff
@@ -340,26 +340,157 @@ def test_get_parts_invalid_dimensions(geom):
     assert np.all(pygeos.equals_exact(np.asarray(geom), pygeos.get_parts(geom)))
 
 
 @pytest.mark.parametrize(
-    "geom", [None, [None], []],
+    "geom",
+    [None, [None], []],
 )
 def test_get_parts_None(geom):
     assert len(pygeos.get_parts(geom)) == 0
 
 
 @pytest.mark.parametrize(
-    "geom", ["foo", ["foo"], 42],
+    "geom",
+    ["foo", ["foo"], 42],
 )
 def test_get_parts_invalid_geometry(geom):
     with pytest.raises(
-        TypeError, match="One of the arguments is of incorrect type.",
+        TypeError,
+        match="One of the arguments is of incorrect type.",
     ):
         pygeos.get_parts(geom)
 
+
+@pytest.mark.skipif(pygeos.geos_version < (3, 6, 0), reason="GEOS < 3.6")
+def test_get_precision():
+    geometries = all_types + (point_z, empty_point, empty_line_string, empty_polygon)
+    # default is 0
+    actual = pygeos.get_precision(geometries).tolist()
+    assert actual == [0] * len(geometries)
+
+    geometry = pygeos.set_precision(geometries, 1)
+    actual = pygeos.get_precision(geometry).tolist()
+    assert actual == [1] * len(geometries)
+
+
+@pytest.mark.skipif(pygeos.geos_version < (3, 6, 0), reason="GEOS < 3.6")
+def test_get_precision_none():
+    assert np.all(np.isnan(pygeos.get_precision([None])))
+
+
+@pytest.mark.skipif(pygeos.geos_version < (3, 6, 0), reason="GEOS < 3.6")
+def test_set_precision():
+    initial_geometry = pygeos.Geometry("POINT (0.9 0.9)")
+    assert pygeos.get_precision(initial_geometry) == 0
+
+    geometry = pygeos.set_precision(initial_geometry, 0)
+    assert pygeos.get_precision(geometry) == 0
+    assert pygeos.equals(geometry, initial_geometry)
+
+    geometry = pygeos.set_precision(initial_geometry, 1)
+    assert pygeos.get_precision(geometry) == 1
+    assert pygeos.equals(geometry, pygeos.Geometry("POINT (1 1)"))
+    # original should remain unchanged
+    assert pygeos.equals(initial_geometry, pygeos.Geometry("POINT (0.9 0.9)"))
+
+
+@pytest.mark.skipif(pygeos.geos_version < (3, 6, 0), reason="GEOS < 3.6")
+def test_set_precision_drop_coords():
+    # setting precision of 0 will not drop duplicated points in original
+    geometry = pygeos.set_precision(
+        pygeos.Geometry("LINESTRING (0 0, 0 0, 0 1, 1 1)"), 0
+    )
+    assert pygeos.equals(geometry, pygeos.Geometry("LINESTRING (0 0, 0 0, 0 1, 1 1)"))
+
+    # setting precision will remove duplicated points
+    geometry = pygeos.set_precision(geometry, 1)
+    assert pygeos.equals(geometry, pygeos.Geometry("LINESTRING (0 0, 0 1, 1 1)"))
+
+
+@pytest.mark.skipif(pygeos.geos_version < (3, 6, 0), reason="GEOS < 3.6")
+def test_set_precision_z():
+    geometry = pygeos.set_precision(pygeos.Geometry("POINT Z (0.9 0.9 0.9)"), 1)
+    assert pygeos.get_precision(geometry) == 1
+    assert pygeos.equals(geometry, pygeos.Geometry("POINT Z (1 1 0.9)"))
+
+
+@pytest.mark.skipif(pygeos.geos_version < (3, 6, 0), reason="GEOS < 3.6")
+def test_set_precision_nan():
```

Comment:
This name clashes with another test.

```suggestion
def test_set_precision_nan_coordinates():
```